relays/client-substrate/src/client.rs

```diff
@@ -192,15 +204,35 @@ impl<C: Chain> Client<C> {
 		Ok(Substrate::<C>::system_account_next_index(&self.client, account).await?)
 	}
 
-	/// Submit an extrinsic for inclusion in a block.
+	/// Submit unsigned extrinsic for inclusion in a block.
 	///
 	/// Note: The given transaction does not need be SCALE encoded beforehand.
-	pub async fn submit_extrinsic(&self, transaction: Bytes) -> Result<C::Hash> {
+	pub async fn submit_unsigned_extrinsic(&self, transaction: Bytes) -> Result<C::Hash> {
 		let tx_hash = Substrate::<C>::author_submit_extrinsic(&self.client, transaction).await?;
 		log::trace!(target: "bridge", "Sent transaction to Substrate node: {:?}", tx_hash);
 		Ok(tx_hash)
 	}
 
+	/// Submit an extrinsic signed by given account.
+	///
+	/// All calls of this method are synchronized, so there can't be more than one active
+	/// `submit_extrinsic_by()` call. This guarantees that no nonces collision may happen
+	/// if all client instances are clones of the same initial `Client`.
+	///
+	/// Note: The extrinsic returned by `prepare_extrinsic` does not need be SCALE encoded beforehand.
```

Comment:
It looks like we stopped SCALE encoding the payload at this layer a while ago, so this comment (and the one for `submit_unsigned_extrinsic` are now outdated.

We should instead have the comment note that we *do* expect the transaction (or result from `prepare_extrinsic` to be SCALE encoded.